Make sure sanitizing from solutions branch doesnt crash with 0 diff

I believe there was a bug before when we tested sanitizing to the master branch. If there was no difference between master and the sanitized solutions sanitizer would crash/break/complain. There was no issue created for this and therefore we should test it to see if it is still a problem.

I think this is the error: 
```
[ERROR] GitCommandError: Cmd('git') failed due to: exit code(1)
  cmdline: git commit -m 'Sanitize files'
  stdout: 'On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean'
``` 

In this case, i got this while trying to test sanitizing my solutions branch. Which just tells me theres no diff and i did something right

--- a/repobee_sanitizer/_sanitize_repo.py
+++ b/repobee_sanitizer/_sanitize_repo.py
@@ -20,6 +20,10 @@ PLUGIN_NAME = "sanitizer"
 LOGGER = daiquiri.getLogger(__file__)
 
 
+class EmptyCommitError(plug.PlugError):
+    pass
+
+
 def check_repo_state(repo_root) -> Optional[str]:
     try:
         repo = git.Repo(repo_root)
@@ -136,7 +140,11 @@ def _git_commit_on_branch(repo_root: pathlib.Path, target_branch: str):
     repo = git.Repo(str(repo_root))
     repo.git.symbolic_ref("HEAD", f"refs/heads/{target_branch}")
     repo.git.add(".", "--force")
-    repo.git.commit("-m", "'Sanitize files'")
+    try:
+        repo.git.commit("-m", "'Sanitize files'")
+    except git.GitCommandError as exc:
+        if "nothing to commit, working tree clean" in exc.__str__():
+            raise EmptyCommitError()
 
 
 def _git_fetch(

--- a/repobee_sanitizer/_sanitize_repo.py
+++ b/repobee_sanitizer/_sanitize_repo.py
@@ -143,8 +143,8 @@ def _git_commit_on_branch(repo_root: pathlib.Path, target_branch: str):
     try:
         repo.git.commit("-m", "'Sanitize files'")
     except git.GitCommandError as exc:
-        if "nothing to commit, working tree clean" in exc.__str__():
-            raise EmptyCommitError()
+        assert "nothing to commit, working tree clean" in str(exc)
+        raise EmptyCommitError() from exc
 
 
 def _git_fetch(

--- a/repobee_sanitizer/sanitizer.py
+++ b/repobee_sanitizer/sanitizer.py
@@ -67,9 +67,17 @@ class SanitizeRepo(plug.Plugin, plug.cli.Command):
             errors = _sanitize_repo.sanitize_files(repo_root, file_relpaths)
         else:
             LOGGER.info(f"Sanitizing repo and updating {self.target_branch}")
-            errors = _sanitize_repo.sanitize_to_target_branch(
-                repo_root, self.target_branch
-            )
+            try:
+                errors = _sanitize_repo.sanitize_to_target_branch(
+                    repo_root, self.target_branch
+                )
+            except _sanitize_repo.EmptyCommitError:
+                return plug.Result(
+                    name="sanitize-repo",
+                    msg="No diff between target branch and sanitized output. "
+                    f"No changes will be made to branch: {self.target_branch}",
+                    status=plug.Status.WARNING,
+                )
 
         if errors:
             return plug.Result(

--- a/tests/test_ext_commands.py
+++ b/tests/test_ext_commands.py
@@ -219,12 +219,12 @@ class TestSanitizeRepo:
             f"--repo-root {fake_repo.path}".split()
         )
 
-        with pytest.raises(plug.PlugError) as exc_info:
-            run_repobee(
-                f"sanitize repo --target-branch master "
-                f"--repo-root {fake_repo.path}".split()
-            )
-            assert "No changes will be made to target branch" in exc_info.msg
+        error = run_repobee(
+            f"sanitize repo --target-branch master "
+            f"--repo-root {fake_repo.path}".split()
+        )
+
+        assert "No changes will be made to branch: master" in error.msg
 
     def test_removes_file_with_shred_marker(self, sanitizer_config, fake_repo):
         """Test that sanitize-repo does not send any files that contain a shred

--- a/tests/test_ext_commands.py
+++ b/tests/test_ext_commands.py
@@ -219,11 +219,12 @@ class TestSanitizeRepo:
             f"--repo-root {fake_repo.path}".split()
         )
 
-        with pytest.raises(git.GitCommandError):
+        with pytest.raises(plug.PlugError) as exc_info:
             run_repobee(
                 f"sanitize repo --target-branch master "
                 f"--repo-root {fake_repo.path}".split()
             )
+            assert exc_info.msg == "No changes will be made to target branch"
 
     def test_removes_file_with_shred_marker(self, sanitizer_config, fake_repo):
         """Test that sanitize-repo does not send any files that contain a shred

--- a/tests/test_ext_commands.py
+++ b/tests/test_ext_commands.py
@@ -224,7 +224,7 @@ class TestSanitizeRepo:
                 f"sanitize repo --target-branch master "
                 f"--repo-root {fake_repo.path}".split()
             )
-            assert exc_info.msg == "No changes will be made to target branch"
+            assert "No changes will be made to target branch" in exc_info.msg
 
     def test_removes_file_with_shred_marker(self, sanitizer_config, fake_repo):
         """Test that sanitize-repo does not send any files that contain a shred

--- a/tests/test_ext_commands.py
+++ b/tests/test_ext_commands.py
@@ -207,6 +207,24 @@ class TestSanitizeRepo:
         fake_repo.repo.git.reset("--hard")
         assert_expected_text_in_files(fake_repo.file_infos)
 
+    def test_throws_error_when_zero_diff_on_target_branch(
+        self, sanitizer_config, fake_repo
+    ):
+        """Test that repobee raises a GitCommandError when we try to sanitize
+        to a branch when there are going to be no changes. To do this, all we
+        have to do is run sanitize twice on the same branch.
+        """
+        run_repobee(
+            f"sanitize repo --target-branch master "
+            f"--repo-root {fake_repo.path}".split()
+        )
+
+        with pytest.raises(git.GitCommandError):
+            run_repobee(
+                f"sanitize repo --target-branch master "
+                f"--repo-root {fake_repo.path}".split()
+            )
+
     def test_removes_file_with_shred_marker(self, sanitizer_config, fake_repo):
         """Test that sanitize-repo does not send any files that contain a shred
         marker to target-branch

